File: Programmers/문제모음/2019 Summer&Winter Coding/멀쩡한사각형.cpp
using namespace std;

int gcd(int a, int b) {
    return b ? gcd(b, a % b) : a;
}

long long solution(int w, int h) {
    long long answer = 1;
    long long total = (long long)h * w;
    int GCD = gcd(w, h);
    
    w /= GCD;
    h /= GCD;
    
    if (h == 1 || w == 1)
        answer = (long long)h * w;
    else
        answer = h + w - 1;
    
    return total - answer * GCD;
}

File: Programmers/문제모음/Summer&Winter Coding (2019)/멀쩡한사각형.java
class Solution {
    public int gcd(int a, int b) {
        return b ? gcd(b, a % b) : a;
    }

    public long solution(int w, int h) {
        long answer = 1;
        long total = (long) w * h;
        int gcd = gcd(w, h);

        w /= gcd;
        h /= gcd;
        
        if (h == 1 || w == 1)
            answer = (long) w * h;
        else
            answer = h + w - 1;

        return total - answer * gcd;
    }
}